### PR TITLE
restrict deck option links

### LINKED ISSUE #16715: Restrict webview in deck options
Webview in deck options needs to be restricted. As of now, you can open the manual and find links to github.com and google.com which can be opened in-app.

This was done before in 2.18 version for SharedDecks webview.

### PR DESCRIPTION
**Pull request by RobozinhoD**
## Purpose / Description
_Describe the problem or feature and motivation_

## Fixes
* Fixes #16715

## Approach
Restrict the view to the manual

## How Has This Been Tested?

[deckoptions.webm](https://github.com/user-attachments/assets/3330b62b-4fb0-43f0-ad4f-ff5bdde9f99f)

## Learning (optional, can help others)
_Describe the research stage_

_Links to blog posts, patterns, libraries or addons used to solve this problem_

## Checklist
_Please, go through these checks before submitting the PR._

- [X] You have a descriptive commit message with a short title (first line, max 50 chars).
- [X] You have commented your code, particularly in hard-to-understand areas
- [X] You have performed a self-review of your own code
- [X] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/anki/pages/PageWebViewClient.kt [hunk 1]
```diff
@@ -38,7 +38,7 @@ open class PageWebViewClient : WebViewClient() {
     /** Wait for the provided promise to complete before showing the WebView */
     open val promiseToWaitFor: String? = null
 
-    var onPageFinishedCallback: OnPageFinishedCallback? = null
+    open var onPageFinishedCallback: OnPageFinishedCallback? = null
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/anki/pages/PageWebViewClient.kt:41]: `open var` is unusual, is there a better way?

**Review by david-allison — APPROVED** [review]

File: AnkiDroid/src/main/java/com/ichi2/anki/pages/PageWebViewClient.kt (hunk 1 above)
Comment (lukstbit) [reply AnkiDroid/src/main/java/com/ichi2/anki/pages/PageWebViewClient.kt:41]: There is a better way by using the original code which used apply {} after creating the PageWebViewClient.

**Review by lukstbit — APPROVED** [review]
Looks fine with a change related to the open property.

**Comment by lukstbit** [comment]
As I said, please keep the original code and use apply {}.

**Comment by RobozinhoD** [comment]
How can I override shouldOverrideUrlLoading with the previous approach and apply {}?

Or do your mean using apply {} instead of in it {}? If so, I'm sorry but I don't see how that's better.

**Comment by lukstbit** [comment]
`object: PageWebViewClient() { }.apply{}`

You're beefing up the changed line count for no reason.

**Comment by RobozinhoD** [comment]
I'm sorry, but I think that apply makes the code less readable.

Is it mandatory to do

**Comment by lukstbit** [comment]
Thank you for making the changes.

**Review by lukstbit — APPROVED** [review]

**Pull request merged**